### PR TITLE
ROC-4095 bank details optional

### PR DESCRIPTION
**Pull request by ravinderpal**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-4095

### Change description ###
Made bank details optional

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 1]
```diff
@@ -613,30 +613,36 @@
...
-        </td>
-      </tr>
-      <tr class="no-border">
-        <td><strong>Balance:</strong></td>
-        <td>
-          {{ bankAccount.balance }}
-        </td>
-      </tr>
+      {% if bankAccounts is defined and bankAccounts is not empty %}
+        {% for bankAccount in bankAccounts | default([]) %}
```
Comment (sabahirfan) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: after the if check, do we need `| default([]` ?

File: domain-model/src/test/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/StatementOfMeansTest.java [hunk 2]
```diff
@@ -125,8 +125,7 @@ public void shouldBeInvalidForEmptyBankAccount() {
         Set<String> errors = validate(statementOfMeans);
         //then
         assertThat(errors)
-                .hasSize(1)
-                .contains("bankAccounts : may not be empty");
+                .doesNotContain("bankAccounts : may not be empty");
```
Comment (sabahirfan) [inline domain-model/src/test/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/StatementOfMeansTest.java]: Please use ` assertThat(response).hasSize(0)` to check there is no error.

Comment (ravinderpal) [reply domain-model/src/test/java/uk/gov/hmcts/cmc/domain/models/statementofmeans/StatementOfMeansTest.java]: Ok 
```suggestion
                .hasSize(0);
```

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html (hunk 1 above)
Comment (ravinderpal) [reply src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: Not necessarly. Removing it.
```suggestion
        {% for bankAccount in bankAccounts %}
```

**Pull request merged**